### DISCUSSION TITLE
After configuring SSO, the login page is not displayed.

### DISCUSSION THREAD

**Discussion by cdwmhcc** [directus/directus#9254] [Archive]
SSO is configured, Github is not displayed on the login page. Did I miss something?

```
####################################################################################################
## SSO (OAuth) Providers

AUTH_PROVIDERS="github,google"

AUTH_GOOGLE_DRIVER="openid"
AUTH_GOOGLE_CLIENT_ID="xxxxxxxxxxxxxxxxxxx"
AUTH_GOOGLE_CLIENT_SECRET= "xxxxxxxxxxxxxxxxxxx"
AUTH_GOOGLE_ISSUER_URL="https://accounts.google.com"
AUTH_GOOGLE_ICON="google"

AUTH_GITHUB_CLIENT_ID="xxxxxxxxxxxxxxx"
AUTH_GITHUB_CLIENT_SECRET="xxxxxxxxxxxxxxxxxxx
AUTH_GITHUB_AUTHORIZE_URL="https://github.com/login/oauth/authorize"
AUTH_GITHUB_ACCESS_URL="https://github.com/login/oauth/access_token"
AUTH_GITHUB_PROFILE_URL="https://api.github.com/user"

####################################################################################################
```

**Comment by aidenfoxx** [comment]
You're missing `AUTH_GITHUB_DRIVER="oauth2"`. Looks like that's missing in the docs, so I'll fix that!

EDIT: Fixed here #9256

**Reply by cdwmhcc** [reply]
Thank you, this is indeed the reason.


But I encountered another problem: `redirect_uri_mismatch` is always prompted:

http://localhost:8081/admin/login?continue&error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23redirect-uri-mismatch

I read this document: https://docs.github.com/en/developers/apps/managing-oauth-apps/troubleshooting-authorization-request-errors

My configuration should be correct, I don't understand why.

![image](https://user-images.githubusercontent.com/8030134/139532175-2d16a162-1b33-41f4-b0b5-7a260d9a15f7.png)

**Reply by aidenfoxx** [reply]
The callback URL should be: "http://localhost:8081/auth/login/github/callback"

The URL you are using is internal to Directus.

**Reply by cdwmhcc** [reply]
Thank you!

But when I click the Github button, the page will refresh and the url becomes the following:
http://localhost:8081/admin/login?reason=INVALID_USER

**Reply by aidenfoxx** [reply]
Do you have a "github" user in Directus? With rc99 we changed authentication so that the users auth provider must match the login method. Assuming you have an existing user you want to log in with you'll need to make sure their `provider` and `external_identifier` are correct. This will vary depending on your configuration (check the docs).

If you create a new user with the github "Provider" and the email address you want to use as "External Identifier", then you should be able to log in.

**Reply by cdwmhcc** [reply]
I am using r100, and there are no github users in the `directus_user` table.

When I click the `Log In with GitHub` button, the page will become blank. After a few seconds, the page will jump to: http://localhost:8081/admin/login?reason=INVALID_USER, and the page will become again log in page.

During this period, no github authorization page appeared.

**Reply by aidenfoxx** [reply]
Exactly! You need a github user in the `directus_users` table.

The reason it shows a white page and redirects back is probably because you're already given Directus permission to authenticate your user, so you don't need to authorize again. The problem is that when you get back to Directus, the OAuth driver can't find a Directus user with both `directus_users.provider = github` and `directus_users.external_identifier = your_github_email`.

**Reply by cdwmhcc** [reply]
Thank you!

But I don't quite understand.When there is no such user in the user table, shouldn't it create a user?

In addition, what should I do if I want to bind a user in the user table to github. It is now gray and cannot be clicked.

**Reply by cdwmhcc** [reply]
![image](https://user-images.githubusercontent.com/8030134/139535379-c89ba144-eff6-4508-89f7-fce89cf6fe61.png)

![image](https://user-images.githubusercontent.com/8030134/139535253-e67f584d-bce1-4b1d-9315-bbfaae613e0c.png)

I see the error printed in the console is still `redirect_uri_mismatch`

![image](https://user-images.githubusercontent.com/8030134/139535241-f69c0e2d-5818-4bfb-9cb3-943a50860eb6.png)

**Reply by aidenfoxx** [reply]
I dunno what the github issue is. You need to make sure you're URLs are correct, and there is only so much I can help with there. Usually Directus runs on port 8055 so not sure why yours is 8081. Make sure your `PUBLIC_URL` is correctly configured.

If you have `ALLOW_PUBLIC_REGISTRATION` enabled it should create the user for you, yes. So you shouldn't have to configure a user manually.

**Reply by cdwmhcc** [reply]
OK,Thank you!

**Reply by cdwmhcc** [reply]
@benhaynes I'm Running Locally.https://docs.directus.io/contributing/running-locally/,help me.

**Reply by aidenfoxx** [reply]
Try setting up Google. They have better error messages.

**Reply by cdwmhcc** [reply]
> Try setting up Google. They have better error messages.

![image](https://user-images.githubusercontent.com/8030134/139538053-a8f1b455-d851-440d-9203-06390cfec9af.png)
![image](https://user-images.githubusercontent.com/8030134/139538096-6539ec60-2e50-4de8-9bda-4e6b3e9a3df4.png)
![image](https://user-images.githubusercontent.com/8030134/139538120-32fa98b6-8450-4498-9669-e942810f1cec.png)

**Reply by aidenfoxx** [reply]
Your `PUBLIC_URL` isn't set up correctly. It needs to be the full URL of your local server

**Reply by cdwmhcc** [reply]
> Your `PUBLIC_URL` isn't set up correctly. It needs to be the full URL of your local server

This is indeed the reason, wow, thank you very much for your help. It worked!

**Reply by cdwmhcc** [reply]
> Your `PUBLIC_URL` isn't set up correctly. It needs to be the full URL of your local server

But after jumping to the verification page and the verification is successful, it will redirect back to the login page without login and no user is created.

http://localhost:8081/admin/login?reason=INVALID_USER

**Reply by cdwmhcc** [reply]
![image](https://user-images.githubusercontent.com/8030134/139538513-a8dae3e0-dd08-4d87-a041-18cf2d19ba5c.png)

**Reply by Jeet-Metapercept** [reply]
Any update on this? Still getting Invalid user creds.

**Directus v9.0.0**

![image](https://user-images.githubusercontent.com/17977232/140727661-e467476a-3704-4d73-a571-942b7205db84.png)

**Reply by aidenfoxx** [reply]
@Jeet-Metapercept If you're having problems it's likely a configuration issue. Please ensure you've checked the docs, and have an appropriately configured use if not using public registration.

The new OAuth system does not work the same as the old one, and you must have a user configured with the correct "Provider" and "External Identifier", the latter of which is controlled by a config option.

**Reply by Jeet-Metapercept** [reply]
Thanks, @aidenfoxx . I am trying to upgrade to the latest version. Before that, I was aware of [breaking changes](https://github.com/directus/directus/releases/tag/v9.0.0-rc.99) and followed the latest docs & updated configuration. 

After a bit more research I found a solution [here](https://github.com/directus/directus/issues/9244#issuecomment-955168246).
